encoding/protobuf/parse.go

```diff
@@ -245,6 +245,9 @@ func (p *protoConverter) addNames(elems []proto.Visitee) {
 		}
 		sym := strings.Join(append(p.path, name), ".")
 		p.symbols[sym] = true
+		qualifiedSym := strings.Join([]string{p.protoPkg, name}, ".")
+		p.symbols[qualifiedSym] = true
+		p.addRef(pos, qualifiedSym, func() ast.Expr { return ast.NewIdent("#" + name) })
```

Comment:
AFAICT you only need to call `addRef`, adding to `symbols` is unnecessary. and potentially harmful, as it is about what symbols the package exports.

I also think adding directly to the scope via `addRef` is a bit weird. Rather than duplicating every name in the scope, what I would do is modify `resolveTopScope` such that when it is looking for `fooBar`, it searches for both `fooBar` (the short name) and `current.package.name.fooBar` (the full name).